scripts/local_cloudbuild_test.py

```diff
@@ -0,0 +1,322 @@
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit test for local_cloudbuild.py"""
+
+import argparse
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+import unittest
+import unittest.mock
+
+import yaml
+
+import local_cloudbuild
+
+
+class ValidationUtilsTest(unittest.TestCase):
+
+    def test_get_field_value(self):
+        valid_cases = (
+            # Normal case, field present and correct type
+            ({ 'present': 1 }, 'present', int, 1),
+            ({ 'present': '1' }, 'present', str, '1'),
+            ({ 'present': [1] }, 'present', list, [1]),
+            ({ 'present': {1: 2} }, 'present', dict, {1: 2}),
+            # Missing field replaced by default
+            ({}, 'missing', str, ''),
+            # Valid conversions
+            ({ 'str_to_int': '1' }, 'str_to_int', int, 1),
+            ({ 'int_to_str': 1 }, 'int_to_str', str, '1'),
+        )
+        for valid_case in valid_cases:
+            with self.subTest(valid_case=valid_case):
+                container, field_name, field_type, expected = valid_case
+                self.assertEqual(
+                    local_cloudbuild.get_field_value(
+                        container, field_name, field_type),
+                    expected)
+
+        invalid_cases = (
+            # Type conversion failures
+            ({ 'bad_list_to_dict': [1] }, 'bad_list_to_dict', dict),
+            ({ 'bad_list_to_str': [1] }, 'bad_list_to_str', str),
+            ({ 'bad_dict_to_list': {1: 2} }, 'bad_dict_to_list', list),
+            ({ 'bad_str_to_int': 'not_an_int' }, 'bad_str_to_int', int),
+            ({ 'bad_str_to_list': 'abc' }, 'bad_str_to_list', list),
+        )
+        for invalid_case in invalid_cases:
+            with self.subTest(invalid_case=invalid_case):
+                container, field_name, field_type = invalid_case
+                with self.assertRaises(ValueError):
+                    local_cloudbuild.get_field_value(
+                        container, field_name, field_type)
+
+    def test_validate_arg_regex(self):
+        self.assertEqual(
+            local_cloudbuild.validate_arg_regex('abc', re.compile('a[b]c')),
+            'abc')
+        with self.assertRaises(argparse.ArgumentTypeError):
+            local_cloudbuild.validate_arg_regex('abc', re.compile('a[d]c'))
+
+
+class LocalCloudbuildTest(unittest.TestCase):
+
+    def setUp(self):
```

Comment:
You can use the pytest tmpdir fixture to do this: http://doc.pytest.org/en/latest/tmpdir.html